capstone-frontend/src/components/MyBooks.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react';
+import { Spinner } from 'react-bootstrap';
+import BookSearchTile from './BookSearchTile';
+
+export class MyBooks extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      books: [],
+      loading: true,
+    }
+  }
+
+  fetchBooks = () => {
+    fetch(`/api/books?userID=${window.localStorage.getItem('userID')}`)
+      .then(response => response.json())
+      .then(bookObjects => Promise.all(bookObjects.map(book => {
+        return fetch(`/api/search?gbookId=${book.gbookID}`)
+          .then(response => response.json())
+          .then(book => book[0])
+      })))
+      .then(books => this._isMounted && this.setState({ books, loading: false }))
```

Comment:
Unfamiliar with `_isMounted` but when I went to read further into it I found this: https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html. Not sure how easy it would be to refactor this, but that link should contain some guidance. If you think it'll take too much time, then not worth doing it now since we're on a time crunch.

Reply:
I remember talking to Geoff about how to make this fix and he said the same thing about unsubscribing from the callbacks. In this case, there's a lot of boilerplate to add to unsubscribe from these fetches and it may take more time than it's worth.

In the website you linked, it states that `isMounted()` (the function) is bad, but suggests `_isMounted` (the variable) like I have here as an easy migration strategy from `isMounted()`, although not the most optimal.

From the site: `An easy migration strategy for anyone upgrading their code to avoid isMounted() is to track the mounted status yourself. Just set a _isMounted property to true in componentDidMount and set it to false in componentWillUnmount, and use this variable to check your component’s status.`

Reply:
Makes sense, nice job considering time/cleanliness tradeoff. Think this is a good solution!